Add exemplar compliance lines to the compliance matrix. (#2103)
open-telemetry/opentelemetry-specification: spec-compliance-matrix.md

| The `View` allows configuring the name description, attributes keys and aggregation of the resulting metric stream.                                                          |          |    |  ?   |    |        |      |        |     |      |     |   -  |       |
| The `View` allows configuring the exemplar reservoir of resulting metric stream.                                                                                             | X        |    |  ?   |    |        |      |        |     |      |     |   -  |       |
| The SDK allows more than one `View` to be specified per instrument.                                                                                                          | X        |    |  ?   |    |        |      |        |     |      |     |   +  |       |

> | The `View` allows configuration of which `ExemplarReservoir` is used to collect exemplars.                                                                                   | X        |    |      |    |        |      |        |     |      |     |      |       |

Reviewer: line 165 above is already having this.

Author: removed

| The metrics exporter `shutdown` function do not block indefinitely.                                                                                                          |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
| The metrics SDK samples `Exemplar`s from measurements.                                                                                                                       |          |    |      |    |        |      |        |     |      |     |      |       |
| Exemplar sampling can be disabled.                                                                                                                                           |          |    |      |    |        |      |        |     |      |     |      |       |
| The metrics SDK samples measurements in the context of a sampled trace by default.                                                                                           |          |    |      |    |        |      |        |     |      |     |      |       |

> | Exemplars retain the full set of `Attributes` associated with the original measurement, regardless of view configuration.                                                    |          |    |      |    |        |      |        |     |      |     |      |       |

Reviewer: ```suggestion
| Exemplars retain any attributes available in the measurement that are not preserved by aggregation or view configuration.                                                    |          |    |      |    |        |      |        |     |      |     |      |       |
```

Borrowing from spec wordings. feel this is better.

| The metrics SDK provides a `SimpleFixedSizeExemplarReservoir` that is used by default for all aggregations except `ExplicitBucketHistogram`.                                 |          |    |      |    |        |      |        |     |      |     |      |       |
| The metrics SDK provides an `AlignedHistogramBucketExemplarReservoir` that is used by default for `ExplicitBucketHistogram` aggregation.                                     |          |    |      |    |        |      |        |     |      |     |      |       |
| The metrics SDK provides an `ExemplarFilter` interface or extension point.                                                                                                   | X        |    |      |    |        |      |        |     |      |     |      |       |
| An `ExemplarFilter` has access to the measurement value, attributes, `Context` and timestamp.                                                                                | X        |    |      |    |        |      |        |     |      |     |      |       |

> | The metrics exporter provides a `shutdown` function.                                                                                                                         |          | +  |  +   |    |        |      |        |     |      |     |   +  |       |
> | The metrics exporter `shutdown` function do not block indefinitely.                                                                                                          |          | +  |  -   |    |        |      |        |     |      |     |   +  |       |

Reviewer 2: Are these lines duplicates of 186-187?
```suggestion
```

Author: Merge conflict auto-resolve failure, looks like.

| The metrics exporter `ForceFlush` can inform the caller whether it succeeded, failed or timed out.                                                                           |          |    |      |    |        |      |        |     |      |     |   +  |       |
| The metrics exporter provides a `shutdown` function.                                                                                                                         |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
| The metrics exporter `shutdown` function do not block indefinitely.                                                                                                          |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
| The metrics SDK can sample `Exemplar`s from measurements.                                                                                                                    |          |    |      |    |        |      |        |     |      |     |      |       |

> | Exemplar sampling can be disabled.                                                                                                                                           |          |    |      |    |        |      |        |     |      |     |      |       |

Reviewer: nit: Since the above says "can sample", it kind of implies that its something that can be enabled/disabled.

Author: True, but I'd rather be explicit about disabling/enabling.   Suggestions for rewording these two?

Reviewer: this is ok.

> | The metrics SDK samples measurements in the context of a sample trace by default.                                                                                            |          |    |      |    |        |      |        |     |      |     |      |       |

Author: Fixed

| Exemplars retain the full set of `Attributes` associated with the original measurement, regardless of view configuration.                                                    |          |    |      |    |        |      |        |     |      |     |      |       |

> | Exemplars contain the associated trace id and span id of the active span in Context when the measurement was taken.                                                          |          |    |      |    |        |      |        |     |      |     |      |       |

Reviewer: ```suggestion
| Exemplars contain the associated trace id and span id of the active span in the Context when the measurement was taken.                                                          |          |    |      |    |        |      |        |     |      |     |      |       |
```

| Exemplars contain the timestamp when the measurement was taken.                                                                                                              |          |    |      |    |        |      |        |     |      |     |      |       |

> | The metrics SDK provides an `ExemplarReservoir` interface or extension point.                                                                                                |          |    |      |    |        |      |        |     |      |     |      |       |

Reviewer: Bit confused with this line and this PR https://github.com/open-telemetry/opentelemetry-specification/pull/2059/files#diff-32d5f289beed96900fa1e6e69a3ee9cc0b6503ce8883672011c44bc772c0fa2aR296.

qn.
Do we allow users to define their own ExemplarReservoir? From the linked PR, its not part of initial release. If thats the case, then this line item in compliance is not needed right?

Author: Ah, I can remove this/mark it as optional.   Yes we're not planning to require SDKs do this for the first release of the specification.

Author: Marked it as optional.  The interface is fully specified and won't change although we aren't requiring a user-configurable extension until Exponential Histograms land.
